<launch>
    <node name="controller" pkg="pid" type="controller" ns="/bbot/left_wheel" output="screen" >
        <param name="node_name" value="left_wheel_pid" />
        <param name="Kp" value="-0.95" />
        <param name="Ki" value="-2.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="10.23" />
        <param name="lower_limit" value="-5.23" />
        <param name="windup_limit" value="10.23" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
        <remap from="/bbot/left_wheel/setpoint" to="/bbot/left_wheel/target_velocity" />
        <remap from="/bbot/left_wheel/state" to="/bbot/left_wheel/current_velocity" />
        <remap from="/bbot/left_wheel/control_effort" to="/bbot/left_wheel/pwm" />
    </node>

    <node name="controller" pkg="pid" type="controller" ns="/bbot/right_wheel" output="screen" >
        <param name="node_name" value="right_wheel_pid" />
        <param name="Kp" value="-0.95" />
        <param name="Ki" value="-2.0" />
        <param name="Kd" value="0.0" />
        <param name="upper_limit" value="10.23" />
        <param name="lower_limit" value="-5.23" />
        <param name="windup_limit" value="10.23" />
        <param name="max_loop_frequency" value="100.0" />
        <param name="min_loop_frequency" value="100.0" />
        <remap from="/bbot/right_wheel/setpoint" to="/bbot/right_wheel/target_velocity" />
        <remap from="/bbot/right_wheel/state" to="/bbot/right_wheel/current_velocity" />
        <remap from="/bbot/right_wheel/control_effort" to="/bbot/right_wheel/pwm" />
    </node>
</launch>